### PR TITLE
Switch to Travis CI container based infrastructure

### PR DESCRIPTION
**Pull request by davehunt**
See https://docs.travis-ci.com/user/migrating-from-legacy/#Why-migrate-to-container-based-infrastructure%3F for reasons to migrate.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,9 +2,6 @@ language: python
 python:
 - 2.7
 - 3.5
-
-sudo: "required"
```
Comment (whimboo) [inline .travis.yml]: I thought "sudo" has to be turned of to set its value to false. I cannot find anything about "required". Mind explaining why we need sudo?

File: .travis.yml [hunk 2]
```diff
@@ -63,9 +60,9 @@ matrix:
     - python: 3.5
 
 before_install:
-  - wget -O geckodriver.tar.gz https://github.com/mozilla/geckodriver/releases/download/v0.12.0/geckodriver-v0.12.0-linux64.tar.gz
-  - gunzip -c geckodriver.tar.gz | tar xopf -
-  - chmod +x geckodriver && sudo mv geckodriver /usr/local/bin
+  - wget -O /tmp/geckodriver.tar.gz https://github.com/mozilla/geckodriver/releases/download/v0.12.0/geckodriver-v0.12.0-linux64.tar.gz
```
Comment (whimboo) [inline .travis.yml]: Why cannot we download to the local dir? Or maybe even a sub directory which we could even add to the cache? We should also update to the 0.13.0 release of geckodriver.

File: .travis.yml [hunk 3]
```diff
@@ -63,9 +60,9 @@ matrix:
...
+  - mkdir $HOME/geckodriver && gunzip -c /tmp/geckodriver.tar.gz | tar xopf - -C $HOME/geckodriver
```
Comment (whimboo) [inline .travis.yml]: Why not calling "tar xfv geckodriver.tar.gz" directly?

File: .travis.yml (hunk 1 above)
Comment (davehunt) [inline .travis.yml]: I'm removing this line, the default is false and container based infrastructure.

File: .travis.yml (hunk 2 above)
Comment (davehunt) [inline .travis.yml]: Updating should be in a separate commit. I'd also argue that using cache should be taken care of separately.

File: .travis.yml (hunk 3 above)
Comment (davehunt) [inline .travis.yml]: This patch was more about switching to container based. I'm sure there are many other optimisations we can make independently. If you feel strongly I'll apply your suggestions in this pull, but as separate commits.

File: .travis.yml (hunk 1 above)
Comment (davehunt) [inline .travis.yml]: I think the default also depends on the age of the project, so projects that pre-date container based infrastructure actually default to true.

File: .travis.yml (hunk 2 above)
Comment (whimboo) [inline .travis.yml]: Sounds fine to me.

File: .travis.yml (hunk 3 above)
Comment (whimboo) [inline .travis.yml]: It's a small change and since we already touch this code... otherwise feel free to file a new issue and maybe a PR.
